src/utils/ParseRouteUtils.js

```diff
@@ -532,8 +534,130 @@ function parseRoute(resp: Object, destinationName: string) {
     }));
 }
 
+/**
+ * Return a route information for V2 returned from parseRoute function,
+ * including travelDistance, totalDuration, and routeSummary.
+ *
+ * Example return object:
+ [
+ {  departureTime: '2018-10-22T03:44:19Z',
+    arrivalTime: '2018-10-22T04:01:46Z',
+    directions: [ [Object], [Object], [Object], [Object] ],
+    startCoords: { lat: 42.441603931224435, long: -76.48638788207742 },
+    endCoords: { lat: 42.45662677611252, long: -76.47693624444763 },
+    startName: 'Risley Hall - Shelter',
+    endName: 'Ithaca Commons - Seneca St',
+    boundingBox:
+     { minLat: 42.441596,
+       minLong: -76.490387,
+       maxLat: 42.456818,
+       maxLong: -76.471642 },
+    travalDistance: 1231,
+    totalDuration: 12,
+    numberOfTransfers: 1,
+    routeSummary: [...],
+  },
+ ...
+  ]
+ * @param resp
+ * @param destinationName
+ * @returns {Promise<Array>}
+ */
+async function parseRouteV2(resp: Object, destinationName: string) {
+    const paths = await parseRoute(resp, destinationName);
+    
+    return paths.map((currPath) => {
+        const {
+            arrivalTime,
+            departureTime,
+            directions,
+        } = currPath;
+
+        let travelDistance = 0;
+        directions.forEach((direction) => {
+            travelDistance += direction.travalDistance;
+        });
+
+        travelDistance *= METERS_TO_MILES;
+        currPath.travelDistance = travelDistance;
+
+        let totalDuration = new Date(arrivalTime) - new Date(departureTime);
+        totalDuration /= ONE_MINUTE_IN_MILLISECONDS;
+        currPath.totalDuration = Math.round(totalDuration);
+
+        currPath.startName = currPath.directions[0].name;
+        currPath.endName = destinationName;
+
+        // routeSummary
+
+        /* routeSummaryElement {
+            stopName: String,
+            direction: RouteSummaryDirection?,
+            shouldStayOnBus: Bool,
+        }
+        */
+
+        /* routeSummaryDirection {
+            type: null || 'walk' || 'bus',
+            routeNumber: null || Int,
+        }
+        */
+
+        const routeSummary = [];
+
+        const walkOnlyRoute = directions.length === 0 || !directions[0].routeNumber;
```

Comment:
@zeladada will this be a sufficient check for `walkOnlyRoute`? I removed the for-loop.